dds/DCPS/XTypes/TypeObject.h

```diff
@@ -213,6 +213,11 @@ namespace XTypes {
     {
       std::memcpy(hash, a_hash, sizeof hash);
     }
+
+    bool operator<(const TypeObjectHashId& other) const
+    {
+      return (kind < other.kind) && hash < other.hash;
```

Comment:
Minor comment but this could be `return (kind < other.kind) && (hash < other.hash);` for consistency.